src/pages/WorkloadDetails/WorkloadInfo/WorkloadPodLogs.tsx

```diff
@@ -0,0 +1,224 @@
+                tooltip="Display logs for the selected pod"
+                handleSelect={key => this.setPod(key)}
+                value={this.state.podValue}
+                label={this.props.pods[this.state.podValue!].name}
+                options={this.podOptions!}
+              />
+              <ToolbarDropdown
+                id={'wpl_containers'}
+                nameDropdown="&nbsp;&nbsp;&nbsp;Container"
+                tooltip="Display logs for the selected pod container"
+                handleSelect={key => this.setContainer(key)}
+                value={this.state.containerInfo.container}
+                label={this.state.containerInfo.container}
+                options={this.state.containerInfo.containerOptions!}
+              />
+              <Toolbar.RightContent>
+                <ToolbarDropdown
+                  id={'wpl_duration'}
+                  handleSelect={key => this.setDuration(key)}
+                  value={this.state.duration}
+                  label={DurationOptions[this.state.duration]}
+                  options={DurationOptions}
+                  tooltip={'Time range for graph data'}
+                />
+                <span style={{ paddingLeft: '0.5em' }}>
+                  <Button id={'wpl_refresh'} disabled={!this.state.podLogs} onClick={() => this.handleRefresh()}>
+                    <Icon name="refresh" />
+                  </Button>
+                </span>
+              </Toolbar.RightContent>
+            </Toolbar>
+            <textarea
+              style={{
+                width: '100%',
+                height: '100%',
+                overflow: 'auto',
+                resize: 'vertical',
+                color: '#fff',
+                backgroundColor: '#003145'
+              }}
+              readOnly={true}
+              value={this.state.podLogs ? this.state.podLogs.logs : 'Loading logs...'}
+            />
+          </>
+        )}
+        {this.state.loadingPodLogsError && <div>{this.state.loadingPodLogsError}</div>}
+      </>
+    );
+  }
+
+  private setPod = (podValue: string) => {
+    const pod = this.props.pods[Number(podValue)];
+    const containerInfo = this.getContainerInfo(pod);
+    this.setState({ containerInfo: containerInfo, podValue: Number(podValue) });
+  };
+
+  private setContainer = (container: string) => {
+    this.setState({
+      containerInfo: { container: container, containerOptions: this.state.containerInfo!.containerOptions }
+    });
+  };
+
+  private setDuration = (duration: string) => {
+    this.setState({ duration: duration });
+  };
+
+  private handleRefresh = () => {
+    const pod = this.props.pods[this.state.podValue!];
+    this.fetchLogs(this.props.namespace, pod.name, this.state.containerInfo!.container, Number(this.state.duration));
+  };
+
+  private getContainerInfo = (pod: Pod): ContainerInfo => {
+    const containers: string[] = [];
+    if (pod.containers) {
+      pod.containers.forEach(c => {
```

Comment:
The `forEach` and `push` is an anti-pattern (mutable and not as direct as `const`... `map` immutable structure)

Reply:
Ok, Thanks for the tip, will change .